src/components/tutorial-meta/index.tsx

```diff
@@ -22,13 +23,21 @@ interface TutorialMetaProps {
 	tutorialId: TutorialData['id']
 }
 
+function getVariantPath(path, variantType) {
+	const { hash, pathname, search } = new URL(
+		path,
+		'https://developer.hashicorp.com'
+	)
+
+	return pathname + search + hash
+}
```

Comment:
The `getTutorialUrl` function will increase in complexity with this work. With state management, that function will have to check if the cookie for the variant is there before fully rewriting the link.